documentation/decisions/003_component_api_style.md

```diff
@@ -0,0 +1,44 @@
+---
+status: Accepted
+date: 2023-02-03
+deciders: stephenjwatkins, OskiTheCoder, kevinalexliu
+---
+
+# Design component APIs based on common patterns that exist across popular libraries.
```

Comment:
during the spike review, we discussed also having polaris as a reference, should i include that here?

Reply:
i personally don't think we need to mention Polaris by name just yet, even though i agree that they are a reference of inspiration, but i think eventually we can have an Inspiration catalog somewhere in the repo once we have a better feel about that